File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/ServerTypes/GrandSubjectDB/GrandSubjectForm.ts
import { StringEditor, TextAreaEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface GrandSubjectForm {
    Title: StringEditor;
    Description: TextAreaEditor;
}

export class GrandSubjectForm extends PrefixedContext {
    static readonly formKey = 'GrandSubjectDB.GrandSubject';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!GrandSubjectForm.init)  {
            GrandSubjectForm.init = true;

            var w0 = StringEditor;
            var w1 = TextAreaEditor;

            initFormType(GrandSubjectForm, [
                'Title', w0,
                'Description', w1
            ]);
        }
    }
}

File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/GrandSubjectDB/GrandSubject/GrandSubjectForm.cs
using Serenity.ComponentModel;
using System;

namespace CorrespondenceSystem.GrandSubjectDB.Forms;

[FormScript("GrandSubjectDB.GrandSubject")]
[BasedOnRow(typeof(GrandSubjectRow), CheckNames = true)]
public class GrandSubjectForm
{

    public string Title { get; set; }
    [Insertable(true)]
    [Updatable]
    [TextAreaEditor]
    public string Description { get; set; }

}